sympy/physics/mechanics/linearize.py

```diff
@@ -83,8 +83,14 @@ def __init__(self, f_0, f_1, f_2, f_3, f_4, f_c, f_v, f_a, q, u,
         # qd and u vectors have any intersecting variables, this can cause
         # problems. We'll fix this with some hackery, and Dummy variables
         dup_vars = set(self._qd).intersection(self.u)
-        self._qd_dup = Matrix([var if var not in dup_vars else Dummy()
-            for var in self._qd])
+        mat_list = []
+        for var in self._qd:
+            if var not in dup_vars:
+                mat_list.append(var)
+            else:
+                print('dummy')
+                mat_list.append(var.as_dummy())
+        self._qd_dup = Matrix(mat_list)
```

Comment:
I thought this might be the issue in the linearizer, but it isn't as the else path isn't hit for the failing example.

Reply:
But this fix should stay in so that the Dummy has the same assumptions as the Symbol.